### ISSUE TITLE
Example requests with `application/x-www-form-urlencoded`

### ISSUE THREAD

**Issue by mzahirr** [scalar/scalar#1451]
### What happens?

In the OpenAPI document, there is a requestBody definition for application/x-www-form-urlencoded type. However, when creating a curl request adhering to this definition, it has been observed that the data appears to be empty when sent via the --data parameter. This issue seems to stem from the expected data format not being received properly, despite creating the curl request in the correct format.

Could you please assist in resolving this issue?

![data](https://uploads.linear.app/06a32804-030f-45a1-98aa-79d02a076af5/e5d96847-bc76-475a-8c0c-f2b40ec74aa8/f6b46d44-834b-4991-b2bc-310bb70e1f3f?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzA2YTMyODA0LTAzMGYtNDVhMS05OGFhLTc5ZDAyYTA3NmFmNS9lNWQ5Njg0Ny1iYzc2LTQ3NWEtOGMwYy1mMmI0MGVjNzRhYTgvZjZiNDZkNDQtODM0Yi00OTkxLWIyYmMtMzEwYmI3MGUxZjNmIiwiaWF0IjoxNzEzODgxNDY1LCJleHAiOjMzMjg0NDQxNDY1fQ.H9ukDuPhSCKWnRYQIY-AC8PlRwQZecHBZmd0v9X6Nds "=1429x")

### What did you expect to happen?

Reviewed the requestBody definition in the OpenAPI document.
Created the curl request correctly, but observed that the data is empty.
If possible, could you explain the reason for this issue and provide a solution?

### How can we reproduce the issue?

1)Reviewed the requestBody definition in the OpenAPI document.
2)Created the curl request correctly, but observed that the data is empty.

### Swagger/OpenAPI Example

```
post:
  tags:
    - "Form"
  summary: Add new question
  description: Add new question to specified form.
  operationId: formQuestions
  requestBody:
    content:
      application/x-www-form-urlencoded:
        schema:
          $ref: '#/components/schemas/Questions'
```

**Comment by mzahirr** [comment]
If I select **application/json** type everything is ok. But I selected   application/x-www-form-urlencoded type data is empty and if I try to test request there is no any parameter.

**Comment by hanspagel** [comment]
Oops, you’re so right. Thanks for reporting! I think the code is just lacking there. Any chance you could share your whole specification with us? Would be fun to work on an API supporting XML when adding these features.

**Comment by mzahirr** [comment]
Hi @hanspagel , sure here is my example. 

`  /form/{formID}/questions:
    post:
      tags:
        - "Form"
      summary: Add new question to specified form
      description: Add new question to specified form. Form questions might have various properties. Examples Is it required? Are there any validations such as 'numeric only'?
      operationId: formQuestions
      requestBody:
        content:
          application/x-www-form-urlencoded:
            schema:
              $ref: '#/components/schemas/Questions'
      parameters:
        - name: apiKey
          in: query
          description: apiKey
          required: true
          explode: true
          schema:
            type: string
            default: 1111111111111
        - name: formID
          in: path
          description: formID
          required: true
          explode: true
          schema:
            type: string
            default: 111111111111
      responses:
        '200':
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/200'
        '404':
          description: Not found
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/404_submission'
        '401':
          description: Not authorized
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/401'`

**Comment by mzahirr** [comment]
For example, the same request is available in Postman, and the curl output is as follows.

`curl --location 'https://******************/API/form/1111111111111/questions?apiKey=1111111111111' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'question%5Btype%5D=control_head' \
--data-urlencode 'question%5Btext%5D=Header' \
--data-urlencode 'question%5Border%5D=1' \
--data-urlencode 'question%5Bname%5D=clickTo'`

**Comment by hanspagel** [comment]
Ah, it’s not really about XML, it’s more about the `application/x-www-form-urlencoded` content type. I’ve submitted a PR to fix this: #2945 :)
